test/Microsoft.AspNetCore.Mvc.Formatters.Xml.Test/XmlDataContractSerializerInputFormatterTest.cs

```diff
@@ -188,8 +190,88 @@ public void HasProperSuppportedEncodings()
             var input = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" +
                                 "<TestLevelOne><SampleInt>" + expectedInt + "</SampleInt>" +
                                 "<sampleString>" + expectedString + "</sampleString></TestLevelOne>";
-
+#pragma warning disable CS0618
             var formatter = new XmlDataContractSerializerInputFormatter(suppressInputFormatterBuffering: true);
+#pragma warning restore CS0618
+
+            var contentBytes = Encoding.UTF8.GetBytes(input);
+            var httpContext = new DefaultHttpContext();
+            httpContext.Features.Set<IHttpResponseFeature>(new TestResponseFeature());
+            httpContext.Request.Body = new NonSeekableReadStream(contentBytes);
+            httpContext.Request.ContentType = "application/xml";
+            var context = GetInputFormatterContext(httpContext, typeof(TestLevelOne));
+
+            // Act
+            var result = await formatter.ReadAsync(context);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.False(result.HasError);
+            var model = Assert.IsType<TestLevelOne>(result.Model);
+
+            Assert.Equal(expectedInt, model.SampleInt);
+            Assert.Equal(expectedString, model.sampleString);
+
+            // Reading again should fail as buffering request body is disabled
+            await Assert.ThrowsAsync<XmlException>(() => formatter.ReadAsync(context));
+        }
+
+        [Fact]
+        public async Task BuffersRequestBody_ByDefaultUsinMvcOptions()
+        {
+            // Arrange
+            var expectedInt = 10;
+            var expectedString = "TestString";
+
+            var input = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" +
+                                "<TestLevelOne><SampleInt>" + expectedInt + "</SampleInt>" +
+                                "<sampleString>" + expectedString + "</sampleString></TestLevelOne>";
```

Comment:
Don't propagate this weird indentation to new tests; it's not consistent even within the file. But, focus on the new tests; no need to clean up the existing extra indentation in this file or `XmlSerializerInputFormatterTest`.